### DISCUSSION TITLE
Link to needs in dynamic functions

### DISCUSSION THREAD

**Discussion by Jattin** [useblocks/sphinx-needs#439] [General]
Hi! I have a use case where i need to specify the requirements using a parameter.

```
.. par-dim:: Max length
    :id: DIM_1
    :value: 100
    :unit: m
    Definition:
       Max length that can be returned.
```

In the documentation, I want to use that parameter to state a requirement.
Expected:
`The returned value should never exceed the length of 100 m.`

So, I used the dynamic function to do that
`The returned value should never exceed the length of [[get_param(id="DIM_1")]].`

```
def get_param(app, need, needs, *args, **kwargs):
     param = needs[kwargs["id"]]
    return "{} {}".format(param["value"], param["unit"])
```

Now, if I want to link to the parameter, I have to change the code to 
`The returned value should never exceed the length of [[get_param(id="DIM_1")]] (see :need: DIM_1).`

I wanted to ask whether this would be possible to return the link also in the dynamic function?
Basically, I want the dynamic function to return `100 m (see <link>)`.

**Comment by PhilipPartsch** [comment]
Hi,
I like your Idea, but I do not like to write such complex code in a rst file.
So I implemented a role for this job.
See Pull Request #468

**Comment by danwos** [comment]
I like the idea of @Jattin, but I'm not sure if it is easily technical possible.
Right now the return value of a dynamic function must be a string, which gets **not** interpreted as rst value.
So a returned ``**value**`` is printed as ``**value**`` and not as **value**.

But maybe to support the above use case, we do not need to use a dynamic function.

It's feels like a better ``:need:`` [role](https://sphinxcontrib-needs.readthedocs.io/en/latest/roles.html#need) could do it, if it has access to the need-values.
Right now you can use ``:need:`` in two ways:
1. ``:need:`DIM_1` ``
2. ``:need:`My own text <DIM_1>` ``

I would like to enhance the "string/naming-feature", to give it access to all the need values:
``:need:`{{value}} {{unit}} of need {{id}} <DIM_1>` ``

Which would create a link with the text: ``100 m of need DIM_1``.

Would this be helpful?

**Reply by PhilipPartsch** [reply]
Why would you use `{{` to provide a option?

**Reply by danwos** [reply]
Sphinx-Needs often use jinja for string templating, therefore the `{{`.
But it could be replaced with anything else.

**Comment by PhilipPartsch** [comment]
After a protoype in PR #611 we make following decisions:
- We want to go with `[[` and `]]` to indicate start and end of option to be fetched. 
- We have to have a follow the python [format ](https://sphinxcontrib-needs.readthedocs.io/en/latest/roles.html#need) handling, as it is how the parameter [needs-role-need-template](https://sphinxcontrib-needs.readthedocs.io/en/latest/configuration.html#needs-role-need-template) works
- As it is described in jinja2 [format ](https://jinja.palletsprojects.com/en/3.1.x/templates/?highlight=format#jinja-filters.format) handling, it is better to use python build in `format` handling
